src/lib/PasswordHasher.js

```diff
@@ -14,24 +14,37 @@
 *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 *    You can download the source here: https://github.com/spark/spark-server
+*
+* @flow
+*
 */
 
-var crypto = require('crypto');
-
-var PasswordHasher = function () {
+import crypto from 'crypto';
 
-};
+const HASH_DIGEST = 'sha1';
+const HASH_ITERATIONS = 30000;
+const KEY_LENGTH = 64;
 
-PasswordHasher.generateSalt = function (callback) {
-	return crypto.randomBytes(64, callback);
-};
+class PasswordHasher {
+  static generateSalt(callback: (error: ?Error, buffer: Buffer) => void) {
+    crypto.randomBytes(64, callback);
+  }
 
-PasswordHasher.hash = function (password, salt, callback) {
-	password = password.toString('base64');
-	salt = salt.toString('base64');
-	return crypto.pbkdf2(password, salt, 30000, 64, function (err, derivedKey) {
-		return callback(err, derivedKey.toString('base64'));
-	});
-};
+  static hash(
+    password: Buffer,
+    salt: Buffer,
+    callback: (error: ?Error, key: string) => void,
+  ) {
+    crypto.pbkdf2(
+      password.toString('base64'),
+      salt.toString('base64'),
```

Comment:
pass and salt here are Buffers, but do we really need making toString() for buffers? seems in some step they transforms automatically and it works the same without toString()..hm..i don't get it yet.